File: challenge-159/RPSLP.js
var readline = require('readline');

var rl = readline.createInterface({
	input: process.stdin,
	output: process.stdout
});

var possibilities = ["rock", "paper", "scissors", "lizard", "spock"];

rl.question("Your input for the game: ", function(answer) {
    var computerChoice = Math.floor(Math.random()*4);
	console.log("Your choice was: " + answer);
    console.log("Computer choice was: " + possibilities[computerChoice]);


	rl.close();
});

File: challenge-159-easy/RPSLP.cpp
#include "RPSLP.h"

string possibilities[5] = {"rock", "paper", "scissors", "lizard", "spock"};

bool RPSLP::validateChoice(string humanInput) {
	bool valid = false;
	for (int i = 0; i < 5; i++) {
		if ( humanInput == possibilities[i]) {
			valid = true;
		}
	}

	return valid;
}

int RPSLP::getWinner(string humanChoice, string computerChoice) {
	if (humanChoice == computerChoice) {
		return 0;
	}

	return 0;
}

string RPSLP::getComputerChoice() {
	int index = rand() % 5 + 1;
	return possibilities[index];
}

void RPSLP::runGame(int rounds) {
	for (int round = 1; round <= rounds; round++) {
		string human, computer = "";
		cout << "Your input for the round " << round << ": ";
		cin >> human;

		if (validateChoice(human)) {
			computer = getComputerChoice();
			cout << "Human choice was: " << human << endl;
			cout << "Computer choice was: " << computer << endl;
		} else {
			cout << "BAH!" << endl;
		}

		cout << endl << "Round " << round << " finished!!" << endl;
	}
}
